src/handler.php

```diff
@@ -145,15 +150,15 @@ public static function switchLocation(\Eluceo\iCal\Component\Event &$e, $oldLoca
      * @param $events
      */
     public static function noDupes(array &$events) {
-        //Sort them
+        //Sort them, first by starttime and then by title
         usort($events, function (Event $a, Event $b) {
             if (strtotime($a->dtstart) > strtotime($b->dtstart)) {
                 return 1;
-            } else if ($a->dtstart > $b->dtstart) {
+            } else if (strtotime($a->dtstart) < strtotime($b->dtstart)) {
```

Comment:
Did you actually check that this isn't a DateTime object? Because it should be :thinking:

Reply:
Unfortunately I haven't found a way to live debug yet. I assume that it is a DateTime object, but mixing string and DateTime comparisons made the function buggy. I took inspiration from the way it was already written (it already contained one cast to string) and just adapted the else condition. I can try comparing both as DateTime, that would probably be a cleaner solution

Reply:
Got it. Lets give it a try :)